homeassistant/components/iskra/sensor.py

```diff
@@ -0,0 +1,343 @@
+        translation_key="phase2_voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        value_func=lambda device: device.measurements.phases[1].voltage.value,
+    ),
+    IskraSensorEntityDescription(
+        key=ATTR_PHASE3_VOLTAGE,
+        translation_key="phase3_voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        value_func=lambda device: device.measurements.phases[2].voltage.value,
+    ),
+    # Current
+    IskraSensorEntityDescription(
+        key=ATTR_PHASE1_CURRENT,
+        translation_key="phase1_current",
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        value_func=lambda device: device.measurements.phases[0].current.value,
+    ),
+    IskraSensorEntityDescription(
+        key=ATTR_PHASE2_CURRENT,
+        translation_key="phase2_current",
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        value_func=lambda device: device.measurements.phases[1].current.value,
+    ),
+    IskraSensorEntityDescription(
+        key=ATTR_PHASE3_CURRENT,
+        translation_key="phase3_current",
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        value_func=lambda device: device.measurements.phases[2].current.value,
+    ),
+    # Frequency
+    IskraSensorEntityDescription(
+        key=ATTR_FREQUENCY,
+        translation_key="frequency",
+        device_class=SensorDeviceClass.FREQUENCY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfFrequency.HERTZ,
+        value_func=lambda device: device.measurements.frequency.value,
+    ),
+    IskraSensorEntityDescription(
+        key=ATTR_CONNECTED_DEVICES,
+        translation_key="connected_devices",
+        value_func=lambda device: len(device.child_devices),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Iskra sensors based on config_entry."""
+
+    # Device that uses the config entry.
+    root_device = entry.runtime_data
+
+    devices = [root_device]
+
+    # Add child devices if the root device is a gateway.
+    if root_device.is_gateway:
+        devices += root_device.get_child_devices()
+
+    # Add sensors for each device.
+    for device in devices:
+        sensors = []
+        coordinator = IskraDataUpdateCoordinator(hass, device)
+        await coordinator.async_config_entry_first_refresh()
```

Comment:
Let's move the creation of coordinators to `__init__.py`

Reply:
fixed